tests/unit/client.py

```diff
@@ -0,0 +1,442 @@
+
+        ])
+        f.write(config)
+        return (tmpdir, cwd)
+
+
+def cleanup((tmpdir, cwd)):
+    """ Cleanup crew, when all tests are done.
+    It will shutdown the haas server,
+    delete any files and folders created for the tests.
+    """
+
+    os.remove('haas.cfg')
+    os.remove('haas.db')
+    os.chdir(cwd)
+    os.rmdir(tmpdir)
+
+
+def initialize_db():
+    """ Creates an  database as defined in haas.cfg."""
+    check_call(['haas-admin', 'db', 'create'])
+
+
+def run_server(cmd):
+    """This function starts a haas server.
+    The arguments in 'cmd' will be a list of arguments like required to start a
+    haas server like ['haas', 'serve', '8888']
+    It will return a handle which can be used to terminate the server when
+    tests finish.
+    """
+    proc = Popen(cmd)
+    return proc
+
+
+def populate_server():
+    """
+    Once the server is started, this function will populate some mock objects
+    to faciliate testing of the client library
+    """
+
+    # Adding nodes, node-01 - node-06
+    url_node = 'http://127.0.0.1:8888/node/'
+    api_nodename = 'http://schema.massopencloud.org/haas/v0/obm/'
+
+    obminfo1 = {
+            "type": api_nodename+'ipmi', "host": "10.10.0.01",
+            "user": "ipmi_u", "password": "pass1234"
+            }
+
+    obminfo2 = {
+            "type": api_nodename+'ipmi', "host": "10.10.0.02",
+            "user": "ipmi_u", "password": "pass1234"
+            }
+
+    obminfo3 = {
+            "type": api_nodename+'ipmi', "host": "10.10.0.03",
+            "user": "ipmi_u", "password": "pass1234"
+            }
+
+    obminfo4 = {
+            "type": api_nodename+'ipmi', "host": "10.10.0.04",
+            "user": "ipmi_u", "password": "pass1234"
+            }
+
+    obminfo5 = {
+            "type": api_nodename+'ipmi', "host": "10.10.0.05",
+            "user": "ipmi_u", "password": "pass1234"
+            }
+
+    obminfo6 = {
+            "type": api_nodename+'ipmi', "host": "10.10.0.06",
+            "user": "ipmi_u", "password": "pass1234"
+            }
+
+    obminfo7 = {
+            "type": api_nodename+'mock', "host": "10.10.0.07",
+            "user": "ipmi_u", "password": "pass1234"
+            }
+
+    obminfo8 = {
+            "type": api_nodename+'mock', "host": "10.10.0.08",
+            "user": "ipmi_u", "password": "pass1234"
+            }
+
+    requests.put(url_node+'node-01', data=json.dumps({"obm": obminfo1}))
+    requests.put(url_node+'node-02', data=json.dumps({"obm": obminfo2}))
+    requests.put(url_node+'node-03', data=json.dumps({"obm": obminfo3}))
+    requests.put(url_node+'node-04', data=json.dumps({"obm": obminfo4}))
+    requests.put(url_node+'node-05', data=json.dumps({"obm": obminfo5}))
+    requests.put(url_node+'node-06', data=json.dumps({"obm": obminfo6}))
+    requests.put(url_node+'node-07', data=json.dumps({"obm": obminfo7}))
+    requests.put(url_node+'node-08', data=json.dumps({"obm": obminfo8}))
```

Comment:
If you make obminfoX a list instead of independent varaibles, you can just do:

```python
for i in range(obminfo):
    requests.put(url_node+('node-%02d' % (i+1)), data=json.dumps(obminfio[i]))
```